controllers/service.go

```diff
@@ -102,18 +102,25 @@ func service(es *egressv1.ExternalService, ready bool, current *corev1.Service)
 		l["egress.monzo.com/hijack-dns"] = "waiting-for-pods"
 	}
 
+	serviceSpec := corev1.ServiceSpec{
+		Selector:        labelsToSelect(es),
+		Ports:           servicePorts(es),
+		SessionAffinity: corev1.ServiceAffinityNone,
+		Type:            corev1.ServiceTypeClusterIP,
+	}
+
+	if es.Spec.ServiceTrafficDistribution != "" {
+		trafficDistribution := es.Spec.ServiceTrafficDistribution
+		serviceSpec.TrafficDistribution = &trafficDistribution
```

Comment:
Do you think it would be helpful to add some validation here for when traffic distribution has an unsupported value?

Reply:
My worry would be if k8s supports extra values in the future we have to also update it here and the failure mode even if we errored here is the same as the operator failing to sync a service manifest (it won't update the svc and logs errors)